File: src/com/designpatterns/creational/singleton/Singleton.java
package com.designpatterns.creational.singleton;

import org.jetbrains.annotations.Contract;

public class Singleton {

    private static Singleton instance;

    @Contract(pure = true)
    private Singleton() {}

    public static synchronized Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }
}

File: src/com/designpatterns/creational/singleton/kotlin/Singleton.kt
package com.designpatterns.creational.singleton.kotlin

import kotlin.jvm.Synchronized

object Singleton {

    @get:Synchronized
    var instance: Singleton? = null
        get() {
            if (field == null) {
                field = Singleton
            }
            return field
        }
        private set
}